### PR TITLE
GDPR-78 Adding retention reasons page

### PR DESCRIPTION
**Pull request by brightonsbox**
Currently this has no functionality, it is just
styling and setting up some basic tests.

### CODE REVIEW

File: prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/specs/RetentionReasonsSpecification.groovy [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    WhereaboutsApi whereaboutsApi = new WhereaboutsApi()
+
+    TestFixture fixture = new TestFixture(browser, elite2api, oauthApi, whereaboutsApi)
+
+    def "should load the retention reasons page"() {
+        setupTests()
+
+        given: "I am on the retention reasons page"
+        to RetentionReasonsPage
+    }
```
Comment (brightonsbox) [inline prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/specs/RetentionReasonsSpecification.groovy]: Currently is just testing that the page loads, no functionality to test yet

File: sass/components/_offender-basic-details.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.bold {
+  font-weight: bold
+}
+
+.basic-details {
+  border-top: 1px solid #b1b4b6;
+  border-bottom: 1px solid #b1b4b6;
+  padding-top: 15px;
+  padding-bottom: 15px;
```
Comment (andrewrlee) [inline sass/components/_offender-basic-details.scss]: you shouldn't really need to add so much custom styling if using gds - there are custom built classes that you can use. https://design-system.service.gov.uk/styles/spacing/

Comment (andrewrlee) [reply sass/components/_offender-basic-details.scss]: similar with typography too

File: views/retentionReasons.njk [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        fieldset: {
+          legend: {
+            text: subtitle,
+            classes: "govuk-fieldset__legend--m"
+          }
+        },
+        hint: {
+          text: instructions 
+        },
+        items: [
```
Comment (andrewrlee) [inline views/retentionReasons.njk]: should these be pulled from the data compliance service?

Comment (andrewrlee) [reply views/retentionReasons.njk]: just thinking that if you wanted to change these that would only be a release to data compliance not here

Comment (brightonsbox) [reply views/retentionReasons.njk]: 👍 yes that's the end goal, but just wanted to start with something that visually matches what I need

File: sass/components/_offender-basic-details.scss (hunk 2 above)
Comment (JamieDraperUK) [reply sass/components/_offender-basic-details.scss]: Be worth looking at the other pages Jon.

Comment (brightonsbox) [reply sass/components/_offender-basic-details.scss]: This is matching Pathfinder, so for consistency wanted to keep it as close to that as possible.  I'll take a look to reduce this though.

File: views/retentionReasons.njk [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                  src="/app/images/{{ offenderBasics.offenderNo }}/data"
+                  alt="Headshot of {{ offenderBasics.firstName }} {{ offenderBasics.lastName }}"
+                  style="float:left"
+                  height="110"
+                  width="90"
+                  data-qa="offender-image"
+          />
+        </div>
+  
+        <div>
```
Comment (JamieDraperUK) [inline views/retentionReasons.njk]: In order to use the correct font here you'd need to use <div class="govuk-body">govuk-body</div>

**Pull request merged**